From patchwork Mon Aug 27 12:47:21 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Li Qiang <liq3ea@gmail.com>
X-Patchwork-Id: 10577061
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 69473139B
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 27 Aug 2018 12:48:08 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 386D029463
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 27 Aug 2018 12:48:07 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 2C62229481; Mon, 27 Aug 2018 12:48:07 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 1A11429463
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 27 Aug 2018 12:48:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726893AbeH0Qd7 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 27 Aug 2018 12:33:59 -0400
Received: from mail-pf1-f196.google.com ([209.85.210.196]:39748 "EHLO
        mail-pf1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726809AbeH0Qd7 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 27 Aug 2018 12:33:59 -0400
Received: by mail-pf1-f196.google.com with SMTP id j8-v6so7704383pff.6;
        Mon, 27 Aug 2018 05:47:28 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=CWQdb4PUOssyyJvHBQOSqSYqNSh0j6gj8Aqj/gw0/20=;
        b=iO4QD1Ibko0bBz6aI3lmWYQs41k7EWHctARh09BUnCYjGBDWPJjxDYmhPaNeVQAYGT
         MpvJEQt1sQGftvmnyybJp84SdBOo85BAYBjBUbvGWxX28jl1bMtWFXXbju0afxxzcadp
         OKS/CGYqx+U1GD05zJ7YjXeByT8JkvxZKCnnQMhNaoU+emqk9dEx4fFGzhM/Y82yihbY
         WBDb+Agf2dN/gAvwBmltW5PfI83JgxulL+fjCYhUlgmy/XBVRyxgJAODyVZS3pM7YAtX
         84uTVA8lffJAZKsK6b5551bN8v0TqOF9oVluPVroAbnDcy3SN/3/xLU2VsgnOzu1YLbi
         sSKQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=CWQdb4PUOssyyJvHBQOSqSYqNSh0j6gj8Aqj/gw0/20=;
        b=eAZ7K2CLizrWYhlM9ilmN/HUZjfnX0dtZTz4lb38kFi/0qDNqhbMkqwfPz/f63YfAb
         h43jiKbWTrCnzN5dkum7qr1k/Cvbl5hDRQqMANGGqCCCWADfGOYLPh3blIBwSWQM+uzh
         hXaecrr5XD+bm4Jqvb/Sib9IK5aepkl/uYBM/Sues5AACDDAjUO8122n1VRT7eQmwA5e
         yTNogsHHbPGiTTDgaA2ijbDt0otlyhIv4kJ1kdhulRGtLeDrIi7/AnM4Jle41idu/ZHB
         KRlPkfAl7RqsRIIwaz9WPC+R2YXk8VepWcIRGGgZN1hlrVUQKMKorf97vQ2nCxJ7kl4r
         NR4Q==
X-Gm-Message-State: APzg51Di9Z4em+k6j1eP2KvnHL7nKcUltNFhgBBSRl4s6vEnDDkduODc
        jUqkVSeuLCH+Gpq8ZJzzgcg=
X-Google-Smtp-Source: 
 ANB0VdbQp7cWJs70wIOGqPiOJkbKfaIVP44RqGfvlkmnj8APIoy/tL+VrWsf3nHZi8poQu2JHxaDvQ==
X-Received: by 2002:a63:6054:: with SMTP id
 u81-v6mr12398606pgb.433.1535374048681;
        Mon, 27 Aug 2018 05:47:28 -0700 (PDT)
Received: from localhost.localdomain.localdomain ([103.65.40.43])
        by smtp.gmail.com with ESMTPSA id
 z6-v6sm19942639pgr.41.2018.08.27.05.47.26
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 27 Aug 2018 05:47:28 -0700 (PDT)
From: Li Qiang <liq3ea@gmail.com>
To: alex.williamson@redhat.com, eric.auger@redhat.com
Cc: kvm@vger.kernel.org, linux-kernel@vger.kernel.org,
        liqiang02@corp.netease.com, Li Qiang <liq3ea@gmail.com>
Subject: [PATCH] vfio: fix potential memory leak in vfio_msi_cap_len
Date: Mon, 27 Aug 2018 05:47:21 -0700
Message-Id: <1535374041-3551-1-git-send-email-liq3ea@gmail.com>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Free the vdev->msi_perm in error path.

Signed-off-by: Li Qiang <liq3ea@gmail.com>
Reviewed-by: Eric Auger <eric.auger@redhat.com>
---
 drivers/vfio/pci/vfio_pci_config.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/vfio/pci/vfio_pci_config.c b/drivers/vfio/pci/vfio_pci_config.c
index 115a36f6f403..62023b4a373b 100644
--- a/drivers/vfio/pci/vfio_pci_config.c
+++ b/drivers/vfio/pci/vfio_pci_config.c
@@ -1180,8 +1180,10 @@ static int vfio_msi_cap_len(struct vfio_pci_device *vdev, u8 pos)
 		return -ENOMEM;
 
 	ret = init_pci_cap_msi_perm(vdev->msi_perm, len, flags);
-	if (ret)
+	if (ret) {
+		kfree(vdev->msi_perm);
 		return ret;
+	}
 
 	return len;
 }
